File: lib/domain/network/local_database/MovieData.java
import 'package:hive/hive.dart';

part 'movie_data.g.dart';

@HiveType(typeId: 0)
class MovieData {
    @HiveField(0)
    final String title;

    @HiveField(1)
    final String description;

    @HiveField(2)
    final String imageUrl;

    MovieData({
        required this.title,
                required this.description,
                required this.imageUrl,
    });
}

File: lib/domain/models/movie_data.dart
import 'dart:developer';

class MovieData {
  int? albumId;
  int? id;
  String? title;
  String? url;
  String? thumbnailUrl;

  MovieData({this.albumId, this.id, this.title, this.url, this.thumbnailUrl, required description, required imageUrl});

  MovieData.fromJson(Map<String, dynamic> json) {
    albumId = json['albumId'];
    id = json['id'];
    title = json['title'];
    url = json['url'];
    thumbnailUrl = json['thumbnailUrl'];
    log( url = json['url']);
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['albumId'] = albumId;
    data['id'] = id;
    data['title'] = title;
    data['url'] = url;
    data['thumbnailUrl'] = thumbnailUrl;
    return data;
  }
}
